gatherers/service-worker.js

```diff
@@ -19,19 +19,25 @@
 const Gather = require('./gather');
 
 class ServiceWorker extends Gather {
+  setup(options) {
```

Comment:
Why not just create the promise here and save it on this object, return it in beforePageLoad?